httpgrpc/server.go

```diff
@@ -181,23 +270,23 @@ func HandleStream(svr interface{}, serviceName string, desc *grpc.StreamDesc, st
 		}
 
 		tr := HttpTrailer{
+			Code:     int32(codes.OK),
+			Message:  codes.OK.String(),
 			Metadata: asTrailerProto(metadata.Join(str.tr...)),
 		}
-		if st, _ := status.FromError(err); st.Code() != codes.OK {
+		if err != nil {
+			st, _ := status.FromError(err)
+			if st.Code() == codes.OK {
+				// preserve all error details, but rewrite the code since we don't want
+				// to send back a non-error status when we know an error occured
+				stpb := st.Proto()
+				stpb.Code = int32(codes.Internal)
+				st = status.FromProto(stpb)
+			}
 			statProto := st.Proto()
 			tr.Code = statProto.Code
 			tr.Message = statProto.Message
 			tr.Details = statProto.Details
-		} else if err != nil {
-			tr.Code = int32(codes.Internal)
-			tr.Message = "Internal Server Error"
-		}
-
-		// must put something the trailing message so it's size is non-zero
-		// (otherwise, it's envelope cannot indicate a negative number, and
-		// it will be confused for a normal response message)
-		if tr.Message == "" {
```

Comment:
I've reformulated the error handling above to match the unary error handling. And I structured it in a way that the `tr` message can never be empty: if there was an error, it will have a non-zero code to serialize at a minimum; if there was no error, it will have an "OK" message to serialize.